src/tools/auth0/handlers/branding.js

```diff
@@ -11,28 +27,66 @@ export default class BrandingHandler extends DefaultHandler {
   }
 
   async getType() {
-    // in case client version does not support branding
-    if (!this.client.branding || typeof this.client.branding.getSettings !== 'function') {
-      return {};
-    }
+    let branding = {};
 
     try {
-      return await this.client.branding.getSettings();
+      // in case client version does not support branding
+      if (this.client.branding && typeof this.client.branding.getSettings === 'function') {
+        branding = await this.client.branding.getSettings();
+      }
+
+      // in case client version does not custom domains
+      if (this.client.customDomains && typeof this.client.customDomains.getAll === 'function') {
+        const customDomains = await this.client.customDomains.getAll();
+        // templates are only supported if there's custom domains.
+        if (customDomains && customDomains.length) {
+          const payload = await this.client.branding.getUniversalLoginTemplate();
+          branding.templates = [
+            {
+              template: constants.UNIVERSAL_LOGIN_TEMPLATE,
+              body: payload.body
+            }
+          ];
+        }
+      }
+
+      return branding;
     } catch (err) {
-      if (err.statusCode === 404) return {};
-      if (err.statusCode === 501) return {};
+      log.debug(`Error calling branding API, ${err.message}, status code: ${err.statusCode}`);
+      if (err.statusCode === 403) return branding;
+      if (err.statusCode === 404) return branding;
+      if (err.statusCode === 501) return branding;
       throw err;
     }
   }
 
   async processChanges(assets) {
     const { branding } = assets;
 
```

Comment:
I believe you need something like:

```
if (!branding) return;
```

Or else if the `branding` property is absent from assets, the `processChanges` method will choke further down the line.